custom_model_runner/datarobot_drum/resource/components/Python/r_fit/fit.R

```diff
@@ -24,6 +24,69 @@ init <- function(code_dir) {
 }
 
 
+load_data <- function(input_filename){
+    tmp = readChar(input_filename, file.info(input_filename)$size)
+    data <- read.csv(text=gsub("\r","", tmp, fixed=TRUE))
+    data
+}
+
+
+process_data <- function(input_filename, target_filename, target_name, num_rows){
+    set.seed(1)
+    # read X
+    df <- load_data(input_filename)
+    # set num_rows
+    if (num_rows == 'ALL'){
+        num_rows <- nrow(df)
+    } else {
+        num_rows <- as.integer(a)
+    }
+    # handle target
+    if (!is.null(target_filename) || !is.null(target_name)){
+        if (!is.null(target_filename)){
+            y_unsampled = load_data(target_filename)
+            assert( length(colnames(y_unsampled)) == 1)
+            assert( nrow(df) == nrow(y_unsampled))
+            df <- cbind(df, y)
+            target_name <- colnames(y)
+        }
+    df <-  df[!(is.na(df[target_name])) , ]
+    X <-  df[,!(names(df) %in% c(target_name))]
+    X <- X[sample(nrow(X), size=num_rows, replace=TRUE ), ]
```

Comment:
am I understanding correctly that we always resample the data with replacement, even if we use the full dataset? is there a reason for this?

Reply:
In python, we sample with replacement so that if a user asks for 100 rows we don't fail, but instead draw from the data that does exist.